crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -150,10 +150,14 @@ pub struct Task<TxPool, D> {
 impl<TxPool, D> Task<TxPool, D>
 where
     TxPool: ports::worker::TxPool,
-    D: ports::worker::OffChainDatabase,
+    D: ports::worker::OffChainDatabase + Clone,
 {
     fn process_block(&mut self, result: SharedImportResult) -> anyhow::Result<()> {
         let block = &result.sealed_block.entity;
+        let _cloned_db = self.database.clone();
+        #[cfg(feature = "fault-proving")]
+        let da_registry_root_getter = || _cloned_db.da_registry_root();
```

Comment:
Why do you need to create a closure and clone? You can just calculate it here and pass, if you need it

Reply:
this pr still in draft, needs some reworking :)